src/estimate.c

```diff
@@ -44,15 +45,46 @@ estimate_max_spread_var(PlannerInfo *root, Var *var)
 	if (!valid)
 		return INVALID_ESTIMATE;
 
+	/* We create a subtransaction to run the functions below, in case they
+	 * abort. */
+	MemoryContext oldcontext = CurrentMemoryContext;
+	ResourceOwner oldowner = CurrentResourceOwner;
+	BeginInternalSubTransaction(NULL);
+
 	PG_TRY();
 	{
 		max = ts_time_value_to_internal(max_datum, var->vartype);
 		min = ts_time_value_to_internal(min_datum, var->vartype);
+		ReleaseCurrentSubTransaction();
+		/* These two are more of a precaution. Neither the context nor the
+		 * owner should change. */
+		MemoryContextSwitchTo(oldcontext);
+		CurrentResourceOwner = oldowner;
 	}
 	PG_CATCH();
 	{
-		valid = false;
+		const int sqlerrcode = geterrcode();
+
+		/*
+		 * We can deal with the Data Exception category and in the Syntax
+		 * Error or Access Rule Violation category, but if the error is an
+		 * insufficient resources category, for example, an out of memory
+		 * error, we should just re-throw it and not treat it as an invalid
+		 * value.
+		 */
+		if (ERRCODE_TO_CATEGORY(sqlerrcode) != ERRCODE_DATA_EXCEPTION &&
+			ERRCODE_TO_CATEGORY(sqlerrcode) != ERRCODE_SYNTAX_ERROR_OR_ACCESS_RULE_VIOLATION)
```

Comment:
Do we even know that these errors can be thrown by the above functions or is this just a generic copy-paste code block?

I am actually questioning the whole need for a try-catch here because it indicates that something is wrong with our min/max values. Is it a common (and valid) occurrence that the max_datum or min_datum cannot be converted? Is it safe to proceed (with planning) if the values are not OK?

Since this code block didn't do the right thing before and just continued a failed transaction, we should have seen bigger issues if this actually happened in the wild. It doesn't seem to happen in tests since there is no codecov. Again, can't we just remove the try-catch?

Reply:
> Do we even know that these errors can be thrown by the above functions or is this just a generic copy-paste code block?

These were picked specifically to capture syntax errors and range violations.

Reply:
> Do we even know that these errors can be thrown by the above functions or is this just a generic copy-paste code block?
> 
> I am actually questioning the whole need for a try-catch here because it indicates that something is wrong with our min/max values. Is it a common (and valid) occurrence that the max_datum or min_datum cannot be converted? Is it safe to proceed (with planning) if the values are not OK?

Here is a [new PR to check if we can remove the try-catch](https://github.com/timescale/timescaledb/pull/7912). It does not seem to fail our tests at least, but I do not know if it can cause problems in situations not covered by our tests.

Reply:
Something is off. There were no coverage for the changes I make inside this function, so I added a printout to this function and there is nothing printed. It seems it is not used.